### ISSUE TITLE
drmaa library not found

### ISSUE THREAD

**Issue by dnieuw** [DennisSchmitz/Jovian_archive#16]
Apparently we do not have libdrmaa.so.0 available on our system, but it is required when running snakemake.

In my case I tried to fix it by installing the torque-drmaa package `sudo yum install torque-drmaa`

However, I get an OSError:

> Exception in thread Thread-1:
Traceback (most recent call last):
  File "/home/dnieuwenhuijse/Software/anaconda3/envs/Jovian_master/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/dnieuwenhuijse/Software/anaconda3/envs/Jovian_master/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/dnieuwenhuijse/Software/anaconda3/envs/Jovian_master/lib/python3.6/site-packages/snakemake/executors.py", line 987, in _wait_for_jobs
    import drmaa
  File "/home/dnieuwenhuijse/Software/anaconda3/envs/Jovian_master/lib/python3.6/site-packages/drmaa/__init__.py", line 65, in <module>
    from .session import JobInfo, JobTemplate, Session
  File "/home/dnieuwenhuijse/Software/anaconda3/envs/Jovian_master/lib/python3.6/site-packages/drmaa/session.py", line 39, in <module>
    from drmaa.helpers import (adapt_rusage, Attribute, attribute_names_iterator,
  File "/home/dnieuwenhuijse/Software/anaconda3/envs/Jovian_master/lib/python3.6/site-packages/drmaa/helpers.py", line 36, in <module>
    from drmaa.wrappers import (drmaa_attr_names_t, drmaa_attr_values_t,
  File "/home/dnieuwenhuijse/Software/anaconda3/envs/Jovian_master/lib/python3.6/site-packages/drmaa/wrappers.py", line 56, in <module>
    _lib = CDLL(libpath, mode=RTLD_GLOBAL)
  File "/home/dnieuwenhuijse/Software/anaconda3/envs/Jovian_master/lib/python3.6/ctypes/__init__.py", line 348, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /lib64/libdrmaa.so.0: undefined symbol: _Z19drmaa_attrib_lookupPKcj
> 
> Segmentation fault

Github posts describing this issue..:
[link](https://github.com/pygridtools/drmaa-python/issues/18)
[link2](https://github.com/cBio/cbio-cluster/issues/168)
Something with the updated library not being compatible with the old python package, but I couldn't figure out how to fix it yet.

Do you know if it is possible to run snakemake without drmaa?

Cheers,

David

**Comment by dnieuw** [comment]
Leaving out the DRMAA options in the snakemake profile.yaml seems to work. You might want some documentation on how to set up the snakemake profile. Or indicate that users should figure out how set it up themselves since it depends on their hardware.

Cheers,

David

**Comment by DennisSchmitz** [comment]
Good point, it's indeed only for grid-engines.

@thierryjanssens , didn't you also have an error requiring you to install torque?

**Comment by thierryjanssens** [comment]
@DennisSchmitz meanwile I have moved to the grid.

**Comment by DennisSchmitz** [comment]
We've found the issue. Currently, the pipeline is setup to work on our grid computer. We specify these settings with the `profile/config.yaml` file, however, if you run these default options on a non-grid computer you will get the above error.

We'll work on a fix, but basically it means having multiple config files: one for grid computers, one for standalone computers. Solution will be pushed in the next version.

**Comment by DennisSchmitz** [comment]
While we are working on a more user-friendly toggle to switch between grid-computer and standalone-computer configurations, below a description of how to reconfigure Jovian to run on a standalone-computer:

- go to the `profile/` folder
- open `config.yaml` and remove the lines containing:
   ```drmaa: " -q bio -n {threads} -R \"span[hosts=1]\""```
   and
   ```drmaa-log-dir: logs/drmaa```
- Lastly, please correct the line containing `cores: 120` with a number suitable to your computer (i.e. the maximum number of computing threads the pipeline may use on your hardware).

The contents of the file should then look something like (N.B. update the number of cores according to your available hardware)...
```
local-cores: 4
cores: 16
config:
   sample_sheet=sample_sheet.yaml
latency-wait: 60
use-conda: true
jobname: Jovian_{name}.jobid{jobid}
```

**Comment by florianzwagemaker** [comment]
This has been fixed in the upcoming `v0.9.5-dev` branch with the addition of a computing mode selector
